octobot_trading/personal_data/orders/order.py

```diff
@@ -95,6 +95,17 @@ def __init__(self, trader, side=None):
         # order state is initialized in initialize_impl()
         self.state = None
 
+        # order activity
+        self.is_active = True   # When is_active=False order is not pushed to exchanges
+        self.active_trigger_price: decimal.Decimal = None    # price threshold from which the order becomes active
+        # when True, order becomes active when current price >= active_trigger_price
+        self.active_trigger_above: bool = None
+        self._active_trigger_event: asyncio.Event = None    # will be set when the price is hit
+        # waiter that will call on_active_trigger() when active_trigger_event is set
+        self._active_trigger_task: asyncio.Task = None
+        # True when a transition between active and inactive is being made
+        self.is_in_active_inactive_transition = False
```

Comment:
I think we should create an ActiveState or something like this to manage all attributes and functions related to order activeness. I think it will be easier to abstract the activeness logic inside a dedicated class.

Reply:
done in `order.active_trigger` using the new `PriceTrigger` class